topic-configuration/README.adoc

```diff
@@ -0,0 +1,689 @@
+|`Unlimited`
+
+h|Supported values
+| `bytes`, `kilobytes`, `megabytes`, `gigabytes`, `terabytes`, `Unlimited`
+
+h|Kafka property name
+|`retention.bytes`
+|===
+--
+
+=== Messages
+
+These properties control how your messages are handled in the Kafka instance.
+
+Maximum message bytes::
++
+--
+Maximum message bytes is the maximum record batch size.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Integer
+
+h|Default value
+|`1 MB`
+
+h|Kafka property name
+|`max.message.bytes`
+|===
+--
+
+Message timestamp type::
++
+--
+Message timestamp type determines whether the timestamp is generated when the message is created (`CreateTime`) or when the message is appended to the log (`LogAppendTime`).
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|String
+
+h|Default value
+|`CreateTime`
+
+h|Kafka property name
+|`message.timestamp.type`
+|===
+--
+
+Maximum message timestamp difference::
++
+--
+Maximum message timestamp difference is the maximum difference allowed between the timestamp specified in the message when it leaves the producer and the timestamp recorded when a broker receives the message.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Long
+
+h|Default value
+|`9223372036854775807 ms` (292271023 years, 2 weeks, 2 days, 13 hours, 12 minutes, 56.064 seconds)
+
+h|Kafka property name
+|`message.timestamp.difference.max.ms`
+|===
+--
+
+Compression type::
++
+--
+Compression type determines the final compression for the topic. The `Producer` compression type retains the original compression set by the producer.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|String
+
+h|Default value
+|`Producer`
+
+h|Kafka property name
+|`compression.type`
+|===
+--
+
+
+Message format version::
++
+--
+Message format version is the `ApiVersion` value that the broker uses to append messages to topics. This value must be a valid `ApiVersion` value, such as 0.8.2, 0.9.0.0, or 0.10.0.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|String
+
+h|Default value
+|`2.7-IV2`
+
+h|Kafka property name
+|`message.format.version`
+|===
+--
+
+
+=== Log
+
+These properties define how your log is handled.
+
+NOTE: Messages are continually appended to the partition log and are assigned their offset.
+
+Cleanup policy::
++
+--
+Cleanup policy determines whether log messages are deleted, compacted, or both. With the `Compact, Delete` option, log segments are first compacted and then deleted according to the retention time or size limit settings.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|Yes
+
+h|Type
+|List
+
+h|Default value
+|`Delete`
+
+h|Supported values
+|`Delete`, `Compact`, `Compact, Delete`
+
+h|Kafka property name
+|`cleanup.policy`
+|===
+--
+
+Delete retention time::
++
+--
+Delete retention time is the amount of time that deletion tombstone markers are retained if the topic is compacted.
```

Comment:
```suggestion
Delete retention time is the amount of time that deletion tombstone markers are retained if the log is compacted.
```

Reply:
Worth introducing with explanation of tombstones? _Producers send a tombstone message to act as a marker to tell a consumer the value is deleted._

Reply:
Done and done. Thanks!